caps/repo.go

```diff
@@ -71,8 +71,8 @@ func (r *Repository) Add(cap *Capability) error {
 	return nil
 }
 
-// HasType checks if the repository contains the given type.
-func (r *Repository) HasType(t *Type) bool {
+// hasType is identical to HasType but doesn't hold the repository lock
```

Comment:
+1 on unlockedFoo when there's a need for having both. I'll kill the public version in this case as you are right it is not really needed.